crates/lib/Cargo.toml

```diff
@@ -21,16 +21,19 @@ required-features = ["commandline"]
 [lib]
 name = "grass"
 path = "src/lib.rs"
-# crate-type = ["cdylib", "rlib"]
+crate-type = ["cdylib", "rlib"]
 bench = false
 
+
 [package.metadata.docs.rs]
 # To build locally:
 # RUSTDOCFLAGS="--cfg doc_cfg" cargo +nightly doc --features=macro --no-deps --open
 features = ["macro"]
 rustdoc-args = ["--cfg", "doc_cfg"]
 
 [dependencies]
+wasm-bindgen = "0.2"
+getrandom = { version = "0.2", features = ["js"] }
```

Comment:
What are these dependencies used for?

Reply:
- `wasm-bindgen` is used for the [JavaScript bindings generation](https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_wasm#compiling_our_code_to_webassembly)
- `getrandom` is a dependency for some libs you use to compile into wasm

Reply:
Hmmm are they still needed if they're not used anywhere in the code? I'm not familiar with wasm-pack, but I would think wasm-bindgen and getrandom are only useful as dependencies if they're actually used in the code.

Regardless, they should be optional dependencies and only enabled for the wasm build. Check out the existing cargo feature, `wasm-exports`

Reply:
This is definitely getting closer! This and the `crate-type` flag are the only two comments I have.

Reply:
As you can see [here](https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_wasm#compiling_our_code_to_webassembly) `wasm-bindgen` is needed to generate the "glue-code JavaScript" for the WebAssembly.

> 2. Runs wasm-bindgen on that WebAssembly, generating a JavaScript file that wraps up that WebAssembly file into a module the browser can understand.

Reply:
Sure, but how does it do that? The `grass_compiler` crate uses wasm-bindgen to generate the [glue code using a macro](https://github.com/connorskees/grass/blob/master/crates/compiler/src/lib.rs#L260). Does wasm-bindgen add something to the build stage that makes it useful regardless of whether it's used in the code directly?

Reply:
But it looks like this is only needed if you use functions from the browser.

Reply:
The `getrandom` dependency however is needed...

https://github.com/JavaScriptPlayground/grass/actions/runs/7883242599/job/21509825027#step:4:1

Reply:
Ah ok for the getrandom dep, it's necessary to manually enable the `js` feature globally.

Reply:
I think you need to enable the wasm-exports feature to make the WASM actually have exports that are accessible. `wasm-exports` is a [cargo feature](https://github.com/connorskees/grass/blob/master/crates/compiler/Cargo.toml#L45) enables use of wasm-bindgen. Right now the WASM compiles, but I don't think it's actually reachable or callable without enabling that feature.

Reply:
Okay I did that. All test are passing and the build is working.